docs/source/changelog.md

```diff
@@ -12,6 +12,326 @@ and as we merge [breaking changes in pull
+
+##### NetworkPolicy egress rules have been extended with a new property [#2508](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2508)
+
+See `Security: breaking change to *.networkPolicy.egress`
+
+##### User scheduler plugin configuration has changed to match `kubescheduler.config.k8s.io/v1beta3` [#2590](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2590)
+
+Advanced customisation of the user scheduler using plugins now requires Kubernetes 1.21+, and the configuration must follow `kubescheduler.config.k8s.io/v1beta3`.
+Customisation is no longer possible with Kubernetes 1.20.
+
+If you are using the user scheduler without custom plugin configuration you are not affected.
+
+##### Kubernetes version 1.20+ is required [#2635](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2635)
+
+This Helm chart uses Kubernetes resources that are not available in Kubernetes versions prior to 1.20.
+
+##### `hub.fsGid` is replaced by `hub.podSecurityContext` [#2720](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2720)
+
+In previous versions of Z2JH `hub.fsGid` set a supplemental group ID, which is required on some K8s systems to ensure JupyterHub has permissions to read/write files on a volume.
+This has been replaced by the more general [`hub.podSecurityContext`](schema_hub.podSecurityContext).
+To upgrade set:
+
+```yaml
+hub:
+  podSecurityContext:
+    fsGroup: GROUP-ID
+```
+
+##### Hub image is based on Debian instead of Ubuntu [#2733](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2733)
+
+The hub container base image has switched from `ubuntu:20.04` to `python:3.9-slim-bullseye` which is based on `debian:bullseye-slim`.
+If you have extended the Z2JH hub image please review the [hub Dockerfile](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/2.0.0/images/hub/Dockerfile).
+Note the singleuser image is not affected.
+
+##### Disabling RBAC requires setting multiple properties,`rbac.enable` is removed [#2736](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2736) [#2739](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2739)
+
+If you previously disabled RBAC using `rbac.enable: False` you should set
+
+````yaml
+rbac:
+  create: False
+hub:
+  serviceAccount:
+    create: false
+proxy:
+  traefik:
+    serviceAccount:
+      create: false
+scheduling:
+  userScheduler:
+    serviceAccount:
+      create: false
+prePuller:
+  hook:
+    serviceAccount:
+      create: false
+
+- The following breaking changes are relevant by bumping various packages:
+  - kubespawner
+    - kubernetes_async
+    - sudo config
+    - not mounting serviceaccount by default?
+  - oauthenticator
+  - jupyterhub
+    - RBAC permissions, so granting admin rights to services isn't recommended etc.
+    - database schema upgrade needed
+
+#### TODO: link out to relevant informational posts
+
+TODO: Write a discourse forum post about the following and highlight that via
+the changelog.
```

Comment:
What's the order for this? Write the post now, then link here, or make a first beta, then write the post, then add link here?

Reply:
Good point, I think getting the beta out first with a link to the changelog and upgrade guide makes sense and will help speed things up. Save the discourse post (and maybe something on the official Jupyter blog) for the final release?

Reply:
```
#### TODO: link out to relevant informational posts

TODO: Write a discourse forum post about the following and highlight that via
the changelog.

- 100+ user-placeholder pods -> node-placeholder pods
- autohttps acme acquisition issues on GKE mitigation
  (notable-dependencies-200)=
```

I think it is quite crucial to be aware of that user placeholder pods shouldn't be over 100 or similar, and that there are currently some issues on GKE with slow setup of networking - which causes failures in the autohttps system. But, lets get this out there now instead and hope that we manage to communicate that sometime else instead.

In [e8cc8a1](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2528/commits/e8cc8a1a141f1e442d690ec7dfb420378f0556e4) I removed this TODO entry.